kernel/src/fs/kernel_config.rs

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::Arc;
+
+use aster_systree::{
+    inherit_sys_branch_node, BranchNodeFields, Error, Result, SysAttrSetBuilder, SysBranchNode,
+    SysObj, SysPerms, SysStr,
+};
+use inherit_methods_macro::inherit_methods;
+use ostd::mm::{VmReader, VmWriter};
+use spin::Once;
+
+/// Registers a new kernel configuration object.
+pub fn register(config_obj: Arc<dyn SysObj>) -> Result<()> {
```

Comment:
I didn't get the purpose of the `kernel_config` module. No one seems to rely on this module.